diff --git a/Makefile b/Makefile
index 77bcbc0..c05dbdd 100644
--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ dwm: ${OBJ}
 	${CC} -o $@ ${OBJ} ${LDFLAGS}
 
 clean:
-	rm -f dwm ${OBJ} dwm-${VERSION}.tar.gz
+	rm -f dwm ${OBJ} dwm-${VERSION}.tar.gz config.h
 
 dist: clean
 	mkdir -p dwm-${VERSION}
diff --git a/config.def.h b/config.def.h
index 1c0b587..9a9e3eb 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,8 +1,10 @@
+#include <X11/XF86keysym.h>
+
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-static const unsigned int borderpx  = 1;        /* border pixel of windows */
-static const unsigned int snap      = 32;       /* snap pixel */
+static const unsigned int borderpx  = 3;        /* border pixel of windows */
+static const unsigned int snap      = 10;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
 static const char *fonts[]          = { "monospace:size=10" };
@@ -19,7 +21,7 @@ static const char *colors[][3]      = {
 };
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char *tags[] = { "Q", "D", "A", "1", "2", "3", "4" };
 
 static const Rule rules[] = {
 	/* xprop(1):
@@ -28,7 +30,9 @@ static const Rule rules[] = {
 	 */
 	/* class      instance    title       tags mask     isfloating   monitor */
 	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
+	{ "Firefox",  NULL,       NULL,       1 << 2,       0,           -1 },
+	{ "htop",     NULL,       NULL,       0,            1,           -1 },
+	{ "mpv",      NULL,       NULL,       0,            1,           -1 },
 };
 
 /* layout(s) */
@@ -38,13 +42,14 @@ static const int resizehints = 1;    /* 1 means respect size hints in tiled resi
 
 static const Layout layouts[] = {
 	/* symbol     arrange function */
-	{ "[]=",      tile },    /* first entry is default */
-	{ "><>",      NULL },    /* no layout function means floating behavior */
+	{ "[T]",      tile },    /* first entry is default */
+	{ "[F]",      NULL },    /* no layout function means floating behavior */
 	{ "[M]",      monocle },
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
+#define ALTKEY Mod1Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
@@ -56,44 +61,71 @@ static const Layout layouts[] = {
 
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
-static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
-static const char *termcmd[]  = { "st", NULL };
+static const char *dmenucmd[]       = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
+static const char *roficmd[]        = { "rofi", "-show", "drun", "-drun-display-format", "{name}", NULL };
+static const char *termcmd[]        = { "xterm", NULL };
+static const char *rangercmd[]      = { "xterm", "-class", "ranger,ranger", "-e", "ranger" };
+static const char *htopcmd[]        = { "xterm", "-geometry", "140x60+0+0", "-class", "htop,htop", "-e", "htop" };
+static const char *monitorcmd[]     = { "/home/bogdan/.config/scripts/monitors.sh", NULL };
+static const char *suspendcmd[]     = { "/home/bogdan/.config/scripts/prompt.sh", "systemctl suspend", NULL };
+static const char *lockcmd[]        = { "/home/bogdan/.config/scripts/lockscreen.sh", NULL };
+static const char *mpvcmd[]         = { "/home/bogdan/.config/scripts/mpv.sh", NULL };
+static const char *printcmd[]       = { "flameshot", "gui", NULL };
+static const char *mutecmd0[]       = { "pactl", "set-sink-mute", "0", "toggle", NULL };
+static const char *mutecmd1[]       = { "pactl", "set-sink-mute", "1", "toggle", NULL };
+static const char *volupcmd[]       = { "amixer", "-q", "set", "Master", "5%+", "unmute", NULL };
+static const char *voldowncmd[]     = { "amixer", "-q", "set", "Master", "5%-", "unmute", NULL };
+static const char *brupcmd[]        = { "light", "-A", "20", NULL };
+static const char *brdowncmd[]      = { "light", "-U", "20", NULL };
 
 static Key keys[] = {
-	/* modifier                     key        function        argument */
-	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
-	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
-	{ MODKEY,                       XK_b,      togglebar,      {0} },
-	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
-	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
-	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
-	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
-	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
-	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
-	{ MODKEY,                       XK_Return, zoom,           {0} },
-	{ MODKEY,                       XK_Tab,    view,           {0} },
-	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
-	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
-	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
-	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
-	{ MODKEY,                       XK_space,  setlayout,      {0} },
-	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
-	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
-	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
-	{ MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
-	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
-	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
-	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
-	TAGKEYS(                        XK_1,                      0)
-	TAGKEYS(                        XK_2,                      1)
-	TAGKEYS(                        XK_3,                      2)
-	TAGKEYS(                        XK_4,                      3)
-	TAGKEYS(                        XK_5,                      4)
-	TAGKEYS(                        XK_6,                      5)
-	TAGKEYS(                        XK_7,                      6)
-	TAGKEYS(                        XK_8,                      7)
-	TAGKEYS(                        XK_9,                      8)
-	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+  /* modifier                     key                               function        argument */
+  { MODKEY,                       XK_r,                             spawn,          {.v = roficmd } },
+  { MODKEY|ShiftMask,             XK_Return,                        spawn,          {.v = termcmd } },
+  { MODKEY|ControlMask,           XK_m,                             spawn,          {.v = monitorcmd } },
+  { MODKEY,                       XK_e,                             spawn,          {.v = rangercmd } },
+  { ControlMask|ShiftMask,        XK_Escape,                        spawn,          {.v = htopcmd } },
+  { ALTKEY|ControlMask,           XK_Delete,                        spawn,          {.v = lockcmd } },
+  { ALTKEY,                       XK_F4,                            spawn,          {.v = suspendcmd } },
+  { ALTKEY|ControlMask,           XK_p,                             spawn,          {.v = mpvcmd } },
+  { 0,                            XK_Print,                         spawn,          {.v = printcmd } },
+  { 0,                            XF86XK_AudioMute,                 spawn,          {.v = mutecmd0 } },
+  { 0,                            XF86XK_AudioMute,                 spawn,          {.v = mutecmd1 } },
+  { 0,                            XF86XK_AudioLowerVolume,          spawn,          {.v = voldowncmd } },
+  { 0,                            XF86XK_AudioRaiseVolume,          spawn,          {.v = volupcmd } },
+  { 0,                            XF86XK_MonBrightnessUp,           spawn,          {.v = brupcmd} },
+  { 0,                            XF86XK_MonBrightnessDown,         spawn,          {.v = brdowncmd} },
+  { MODKEY,                       XK_b,                             togglebar,      {0} },
+  { ALTKEY,                       XK_Tab,                           focusstack,     {.i = +1 } },
+  { ALTKEY|ShiftMask,             XK_Tab,                           focusstack,     {.i = -1 } },
+  { MODKEY,                       XK_j,                             focusstack,     {.i = +1 } },
+  { MODKEY,                       XK_k,                             focusstack,     {.i = -1 } },
+  { MODKEY,                       XK_i,                             incnmaster,     {.i = +1 } },
+  { MODKEY,                       XK_u,                             incnmaster,     {.i = -1 } },
+  { MODKEY,                       XK_h,                             setmfact,       {.f = -0.05} },
+  { MODKEY,                       XK_l,                             setmfact,       {.f = +0.05} },
+  { MODKEY,                       XK_Return,                        zoom,           {0} },
+  { MODKEY,                       XK_grave,                         view,           {0} },
+  { MODKEY,                       XK_w,                             killclient,     {0} },
+  { MODKEY,                       XK_t,                             setlayout,      {.v = &layouts[0]} },
+  { MODKEY,                       XK_f,                             setlayout,      {.v = &layouts[1]} },
+  { MODKEY,                       XK_m,                             setlayout,      {.v = &layouts[2]} },
+  { MODKEY,                       XK_space,                         setlayout,      {0} },
+  { MODKEY|ShiftMask,             XK_space,                         togglefloating, {0} },
+  { MODKEY,                       XK_0,                             view,           {.ui = ~0 } },
+  { MODKEY|ShiftMask,             XK_0,                             tag,            {.ui = ~0 } },
+  { MODKEY,                       XK_comma,                         focusmon,       {.i = -1 } },
+  { MODKEY,                       XK_period,                        focusmon,       {.i = +1 } },
+  { MODKEY|ShiftMask,             XK_comma,                         tagmon,         {.i = -1 } },
+  { MODKEY|ShiftMask,             XK_period,                        tagmon,         {.i = +1 } },
+  TAGKEYS(                        XK_q,                                             0)
+  TAGKEYS(                        XK_d,                                             1)
+  TAGKEYS(                        XK_a,                                             2)
+  TAGKEYS(                        XK_1,                                             3)
+  TAGKEYS(                        XK_2,                                             4)
+  TAGKEYS(                        XK_3,                                             5)
+  TAGKEYS(                        XK_4,                                             6)
+  { MODKEY|ShiftMask,             XK_q,                             quit,           {0} },
 };
 
 /* button definitions */
diff --git a/config.mk b/config.mk
index 6d36cb7..7553b01 100644
--- a/config.mk
+++ b/config.mk
@@ -11,8 +11,8 @@ X11INC = /usr/X11R6/include
 X11LIB = /usr/X11R6/lib
 
 # Xinerama, comment if you don't want it
-XINERAMALIBS  = -lXinerama
-XINERAMAFLAGS = -DXINERAMA
+# XINERAMALIBS  = -lXinerama
+# XINERAMAFLAGS = -DXINERAMA
 
 # freetype
 FREETYPELIBS = -lfontconfig -lXft
diff --git a/patches/bogdbo.diff b/patches/bogdbo.diff
index 6347273..192567c 100644
--- a/patches/bogdbo.diff
+++ b/patches/bogdbo.diff
@@ -1,163 +0,0 @@
-diff --git a/config.def.h b/config.def.h
-index 1c0b587..9a9e3eb 100644
---- a/config.def.h
-+++ b/config.def.h
-@@ -1,8 +1,10 @@
-+#include <X11/XF86keysym.h>
-+
- /* See LICENSE file for copyright and license details. */
- 
- /* appearance */
--static const unsigned int borderpx  = 1;        /* border pixel of windows */
--static const unsigned int snap      = 32;       /* snap pixel */
-+static const unsigned int borderpx  = 3;        /* border pixel of windows */
-+static const unsigned int snap      = 10;       /* snap pixel */
- static const int showbar            = 1;        /* 0 means no bar */
- static const int topbar             = 1;        /* 0 means bottom bar */
- static const char *fonts[]          = { "monospace:size=10" };
-@@ -19,7 +21,7 @@ static const char *colors[][3]      = {
- };
- 
- /* tagging */
--static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
-+static const char *tags[] = { "Q", "D", "A", "1", "2", "3", "4" };
- 
- static const Rule rules[] = {
- 	/* xprop(1):
-@@ -28,7 +30,9 @@ static const Rule rules[] = {
- 	 */
- 	/* class      instance    title       tags mask     isfloating   monitor */
- 	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
--	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
-+	{ "Firefox",  NULL,       NULL,       1 << 2,       0,           -1 },
-+	{ "htop",     NULL,       NULL,       0,            1,           -1 },
-+	{ "mpv",      NULL,       NULL,       0,            1,           -1 },
- };
- 
- /* layout(s) */
-@@ -38,13 +42,14 @@ static const int resizehints = 1;    /* 1 means respect size hints in tiled resi
- 
- static const Layout layouts[] = {
- 	/* symbol     arrange function */
--	{ "[]=",      tile },    /* first entry is default */
--	{ "><>",      NULL },    /* no layout function means floating behavior */
-+	{ "[T]",      tile },    /* first entry is default */
-+	{ "[F]",      NULL },    /* no layout function means floating behavior */
- 	{ "[M]",      monocle },
- };
- 
- /* key definitions */
--#define MODKEY Mod1Mask
-+#define MODKEY Mod4Mask
-+#define ALTKEY Mod1Mask
- #define TAGKEYS(KEY,TAG) \
- 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
- 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
-@@ -56,44 +61,71 @@ static const Layout layouts[] = {
- 
- /* commands */
- static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
--static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
--static const char *termcmd[]  = { "st", NULL };
-+static const char *dmenucmd[]       = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
-+static const char *roficmd[]        = { "rofi", "-show", "drun", "-drun-display-format", "{name}", NULL };
-+static const char *termcmd[]        = { "xterm", NULL };
-+static const char *rangercmd[]      = { "xterm", "-class", "ranger,ranger", "-e", "ranger" };
-+static const char *htopcmd[]        = { "xterm", "-geometry", "140x60+0+0", "-class", "htop,htop", "-e", "htop" };
-+static const char *monitorcmd[]     = { "/home/bogdan/.config/scripts/monitors.sh", NULL };
-+static const char *suspendcmd[]     = { "/home/bogdan/.config/scripts/prompt.sh", "systemctl suspend", NULL };
-+static const char *lockcmd[]        = { "/home/bogdan/.config/scripts/lockscreen.sh", NULL };
-+static const char *mpvcmd[]         = { "/home/bogdan/.config/scripts/mpv.sh", NULL };
-+static const char *printcmd[]       = { "flameshot", "gui", NULL };
-+static const char *mutecmd0[]       = { "pactl", "set-sink-mute", "0", "toggle", NULL };
-+static const char *mutecmd1[]       = { "pactl", "set-sink-mute", "1", "toggle", NULL };
-+static const char *volupcmd[]       = { "amixer", "-q", "set", "Master", "5%+", "unmute", NULL };
-+static const char *voldowncmd[]     = { "amixer", "-q", "set", "Master", "5%-", "unmute", NULL };
-+static const char *brupcmd[]        = { "light", "-A", "20", NULL };
-+static const char *brdowncmd[]      = { "light", "-U", "20", NULL };
- 
- static Key keys[] = {
--	/* modifier                     key        function        argument */
--	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
--	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
--	{ MODKEY,                       XK_b,      togglebar,      {0} },
--	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
--	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
--	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
--	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
--	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
--	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
--	{ MODKEY,                       XK_Return, zoom,           {0} },
--	{ MODKEY,                       XK_Tab,    view,           {0} },
--	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
--	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
--	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
--	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
--	{ MODKEY,                       XK_space,  setlayout,      {0} },
--	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
--	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
--	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
--	{ MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
--	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
--	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
--	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
--	TAGKEYS(                        XK_1,                      0)
--	TAGKEYS(                        XK_2,                      1)
--	TAGKEYS(                        XK_3,                      2)
--	TAGKEYS(                        XK_4,                      3)
--	TAGKEYS(                        XK_5,                      4)
--	TAGKEYS(                        XK_6,                      5)
--	TAGKEYS(                        XK_7,                      6)
--	TAGKEYS(                        XK_8,                      7)
--	TAGKEYS(                        XK_9,                      8)
--	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
-+  /* modifier                     key                               function        argument */
-+  { MODKEY,                       XK_r,                             spawn,          {.v = roficmd } },
-+  { MODKEY|ShiftMask,             XK_Return,                        spawn,          {.v = termcmd } },
-+  { MODKEY|ControlMask,           XK_m,                             spawn,          {.v = monitorcmd } },
-+  { MODKEY,                       XK_e,                             spawn,          {.v = rangercmd } },
-+  { ControlMask|ShiftMask,        XK_Escape,                        spawn,          {.v = htopcmd } },
-+  { ALTKEY|ControlMask,           XK_Delete,                        spawn,          {.v = lockcmd } },
-+  { ALTKEY,                       XK_F4,                            spawn,          {.v = suspendcmd } },
-+  { ALTKEY|ControlMask,           XK_p,                             spawn,          {.v = mpvcmd } },
-+  { 0,                            XK_Print,                         spawn,          {.v = printcmd } },
-+  { 0,                            XF86XK_AudioMute,                 spawn,          {.v = mutecmd0 } },
-+  { 0,                            XF86XK_AudioMute,                 spawn,          {.v = mutecmd1 } },
-+  { 0,                            XF86XK_AudioLowerVolume,          spawn,          {.v = voldowncmd } },
-+  { 0,                            XF86XK_AudioRaiseVolume,          spawn,          {.v = volupcmd } },
-+  { 0,                            XF86XK_MonBrightnessUp,           spawn,          {.v = brupcmd} },
-+  { 0,                            XF86XK_MonBrightnessDown,         spawn,          {.v = brdowncmd} },
-+  { MODKEY,                       XK_b,                             togglebar,      {0} },
-+  { ALTKEY,                       XK_Tab,                           focusstack,     {.i = +1 } },
-+  { ALTKEY|ShiftMask,             XK_Tab,                           focusstack,     {.i = -1 } },
-+  { MODKEY,                       XK_j,                             focusstack,     {.i = +1 } },
-+  { MODKEY,                       XK_k,                             focusstack,     {.i = -1 } },
-+  { MODKEY,                       XK_i,                             incnmaster,     {.i = +1 } },
-+  { MODKEY,                       XK_u,                             incnmaster,     {.i = -1 } },
-+  { MODKEY,                       XK_h,                             setmfact,       {.f = -0.05} },
-+  { MODKEY,                       XK_l,                             setmfact,       {.f = +0.05} },
-+  { MODKEY,                       XK_Return,                        zoom,           {0} },
-+  { MODKEY,                       XK_grave,                         view,           {0} },
-+  { MODKEY,                       XK_w,                             killclient,     {0} },
-+  { MODKEY,                       XK_t,                             setlayout,      {.v = &layouts[0]} },
-+  { MODKEY,                       XK_f,                             setlayout,      {.v = &layouts[1]} },
-+  { MODKEY,                       XK_m,                             setlayout,      {.v = &layouts[2]} },
-+  { MODKEY,                       XK_space,                         setlayout,      {0} },
-+  { MODKEY|ShiftMask,             XK_space,                         togglefloating, {0} },
-+  { MODKEY,                       XK_0,                             view,           {.ui = ~0 } },
-+  { MODKEY|ShiftMask,             XK_0,                             tag,            {.ui = ~0 } },
-+  { MODKEY,                       XK_comma,                         focusmon,       {.i = -1 } },
-+  { MODKEY,                       XK_period,                        focusmon,       {.i = +1 } },
-+  { MODKEY|ShiftMask,             XK_comma,                         tagmon,         {.i = -1 } },
-+  { MODKEY|ShiftMask,             XK_period,                        tagmon,         {.i = +1 } },
-+  TAGKEYS(                        XK_q,                                             0)
-+  TAGKEYS(                        XK_d,                                             1)
-+  TAGKEYS(                        XK_a,                                             2)
-+  TAGKEYS(                        XK_1,                                             3)
-+  TAGKEYS(                        XK_2,                                             4)
-+  TAGKEYS(                        XK_3,                                             5)
-+  TAGKEYS(                        XK_4,                                             6)
-+  { MODKEY|ShiftMask,             XK_q,                             quit,           {0} },
- };
- 
- /* button definitions */
